### PR TITLE
Issue/317 add web UI section to the configuration file iso5

### LINKED ISSUE #317: Add web-ui section to the configuration file
There are currently two issues with the configuration options of the web-console:

* First of all, not all options are part of the same section. All authentication-related options are present in the old `dashboard` section. All other options, like whether the dashboard is enabled or not, is part of the `web-console` section.
* Secondly, the name of the config section of the web-console is too application specific. As such, we have to create a new configuration section every time we introduce a new web-interface.

Tasks:

- [x] Create a new section `web-ui`. This section will contain all config options that are currently part of the `web-console` and the `dashboard` section.
- [ ] Keep the `web-console` and the `dashboard` section, but log a deprecation warning when it's used and mention that the `web-ui` section should be used instead.

Add deprecation note in iso5, drop support in iso6 (with changelog note).

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description
- Move ui related options from inmanta-core to inmanta-ui
- Log a deprecation warning when the web-console or the dashboard sections are used 

part of #317 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Comment by Hugo-Inmanta** [comment]
Tests on jenkins are expected to fail until https://github.com/inmanta/inmanta-core/pull/5077 has been merged in

File: changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso5.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+description: The 'dashboard' and 'web-console' sections are being deprecated. The 'web-ui' section should now be used instead for configuration options related to web interfaces.
```
Comment (arnaudsjs) [inline changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso5.yml:2]:
```suggestion
description: The 'web-console' section is being deprecated. The 'web-ui' section should now be used instead for configuration options related to web interfaces.
```

File: src/inmanta_ui/config.py [hunk 2]
```diff
@@ -16,20 +16,104 @@
     Contact: code@inmanta.com
 """
 from inmanta.config import Option, is_bool, is_str
+from inmanta.server.config import dash_auth_url, dash_client_id, dash_enable, dash_path, dash_realm
 
-web_console_enabled = Option("web-console", "enabled", True, "Should the server should host the web-console or not", is_bool)
+web_console_enabled = Option(
```
Comment (arnaudsjs) [inline src/inmanta_ui/config.py:21]: I like the banner above the dashboard-related configuration options. Let's also add one above the old and the new configuration options related to the web-console.

File: changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso5.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+change-type: minor
+issue-nr: 317
+issue-repo: inmanta-ui
+destination-branches: [iso5]
```
Comment (arnaudsjs) [inline changelogs/unreleased/317-add-web-ui-section-to-the-configuration-file-iso5.yml:6]: This is an invalid PR. The base branch of this PR is `master`, but the changelog entry indicates that it should be merged into `iso5` only.

File: tests/test_web_ui_options.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from inmanta.config import Config
+from inmanta.server import config as opt
+from inmanta_ui import config as cfg
```
Comment (arnaudsjs) [inline tests/test_web_ui_options.py:21]: `opt` and `cfg` are chosen very arbitrarily. I would rename then to make it clear what they refer to:
```suggestion
from inmanta.server import config as opt_core
from inmanta_ui import config as opt_ui
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Comment by Hugo-Inmanta** [comment]
This PR was based off the wrong branch.
New valid PR is here: https://github.com/inmanta/inmanta-ui/pull/336

**Pull request closed without merging**